### ISSUE TITLE
web.files.delete throws TypeError: source.on is not a function

### ISSUE THREAD

**Issue by evil-shrike** [slackapi/node-slack-sdk#467]
### Description

Describe your issue here.

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [x] bug
- [ ] enhancement (feature request)
- [ ] question
- [ ] documentation related
- [ ] testing related
- [ ] discussion

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

---

#### Reproducible in:

`@slack/client` version: 3.16.0
node version: node 9.3
OS version(s): Windows 10

#### Steps to reproduce:

1. according to https://api.slack.com/methods/files.delete I wrote a simple code:
```
const { IncomingWebhook, WebClient } = require('@slack/client');
const TOKEN = "...";
const web = new WebClient(TOKEN);
web.files.delete({file: 'F4545RVL0'})
	.then(console.log)
    .catch(console.error);

```
2. run it and get an error "TypeError: source.on is not a function":
```
C:\Work\Learn\slack>node app.js
C:\Work\Learn\slack\node_modules\delayed-stream\lib\delayed_stream.js:33
  source.on('error', function() {});
         ^

TypeError: source.on is not a function
    at Function.DelayedStream.create (C:\Work\Learn\slack\node_modules\delayed-stream\lib\delayed_stream.js:33:10)
    at FormData.CombinedStream.append (C:\Work\Learn\slack\node_modules\combined-stream\lib\combined_stream.js:43:37)
    at FormData.append (C:\Work\Learn\slack\node_modules\form-data\lib\form_data.js:68:3)
    at appendFormValue (C:\Work\Learn\slack\node_modules\request\request.js:327:21)
    at Request.init (C:\Work\Learn\slack\node_modules\request\request.js:338:11)
    at new Request (C:\Work\Learn\slack\node_modules\request\request.js:128:8)
    at request (C:\Work\Learn\slack\node_modules\request\index.js:54:10)
    at Function.post (C:\Work\Learn\slack\node_modules\request\index.js:62:12)
    at WebAPIClient.requestTransport [as transport] (C:\Work\Learn\slack\node_modules\@slack\client\lib\clients\transports\request.js:76:11)
    at RetryOperation.attemptTransportCall [as _fn] (C:\Work\Learn\slack\node_modules\@slack\client\lib\clients\client.js:129:10)
    at RetryOperation.attempt (C:\Work\Learn\slack\node_modules\retry\lib\retry_operation.js:88:8)
    at WebAPIClient._callTransport (C:\Work\Learn\slack\node_modules\@slack\client\lib\clients\client.js:127:11)
    at wrapper (C:\Work\Learn\slack\node_modules\lodash\lodash.js:4941:19)
    at C:\Work\Learn\slack\node_modules\async\lib\async.js:987:13
    at Immediate.process (C:\Work\Learn\slack\node_modules\async\lib\async.js:953:21)
    at runCallback (timers.js:773:18)
```

#### Expected result:

No exception in any case.

#### Actual result:

What actually happened
Unhandled exception is thrown.

**Comment by clavin** [comment]
So, this error stems due to the arguments of the [`WebClient.files.delete`](https://slackapi.github.io/node-slack-sdk/reference/FilesFacet#FilesFacet+delete) function. To clarify, `WebClient.files.delete` takes *only one* argument, and that argument *is* the file ID--**not** an object with a `file` field.

As a rule of thumb, with this SDK, all required arguments tend to be direct arguments of `WebClient` methods, and all optional arguments are in an `opts` object. See **the note** of [this example](https://slackapi.github.io/node-slack-sdk/web_api#posting-a-message) for more clarification.

Since the file ID is a required argument of the `files.delete` method, the ID *is* an argument of the method, not an option.

**tl;dr:** Call `web.files.delete` with the file ID, not an object.

```
// ...
web.files.delete('F4545RVL0')
    .then(/* ... */)
    .catch(/* ... */);
```
